### PR TITLE
[TypeDeclaration] Add MergeDateTimePropertyTypeDeclarationRector

### LINKED ISSUE #8338: [TypeDeclaratoin] MergeDateTimePropertyTypeDeclarationRectror
During working on @mautic codebase, there came a need for date time interface vs date time types.

I always have to update the property type and remove the docblock manually. Much waste of time on 50+ properties, so we should have a rule for that :slightly_smiling_face: 

```diff
-/**
- * @var DateTimeInterface
- */
-private DateTime $dateTime;
+private DateTimeInterface $dateTime;
```

### PR DESCRIPTION
**Pull request by samsonasik**
Closes https://github.com/rectorphp/rector/issues/8338

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed :tada: @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Looks good to me, thank you :clap:

File: rules-tests/TypeDeclaration/Rector/Class_/MergeDateTimePropertyTypeDeclarationRector/Fixture/some_class.php.inc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Class_\MergeDateTimePropertyTypeDeclarationRector\Fixture;
+
+final class SomeClass
+{
+    private \DateTime $dateTime;
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/Class_/MergeDateTimePropertyTypeDeclarationRector/Fixture/some_class.php.inc:21]: Oops, I forgot to change type to DateTimeInterface

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/Class_/MergeDateTimePropertyTypeDeclarationRector/Fixture/some_class.php.inc:21]: I created new PR for it:

- https://github.com/rectorphp/rector-src/pull/5329

**Pull request merged**